rebasehelper/tags.py

```diff
@@ -111,32 +137,35 @@ def sanitize(name):
                     m = tag_re.match(line)
                     if m:
                         span = cast(Tuple[int, int], tuple(x + start for x in m.span('value')))
-                        result.append(Tag(section, index, sanitize(m.group('name')), span, valid))
+                        result.append(Tag(section_index, section, index, sanitize(m.group('name')), span, valid))
         return cast(Tuple[Tag], tuple(result))
 
     @classmethod
-    def _filter(cls, tags: Tuple[Tag], section: Optional[str] = None, name: Optional[str] = None,
-                valid: Optional[bool] = True) -> Iterator[Tag]:
+    def _filter(cls, tags: Tuple[Tag], section_name: Optional[str] = None, name: Optional[str] = None,
+                valid: Optional[bool] = True, section_index: int = None) -> Iterator[Tag]:
```

Comment:
`section_index` should be `Optional[int]` and it would make more sense if it was the first argument.